OpenRA.Platforms.Default/ThreadedGraphicsContext.cs

```diff
@@ -96,12 +97,26 @@ void RenderThread(object contextObject)
 								context.CreateFrameBuffer(t.Item1, (ITextureInternal)CreateTexture(), t.Item2));
 						};
 					getCreateShader = bindings => new ThreadedShader(this, context.CreateShader((IShaderBindings)bindings));
-					getCreateVertexBuffer =
+					getCreateEmptyVertexBuffer =
 						tuple =>
 						{
 							(object t, var type) = ((int, Type))tuple;
-							var vertexBuffer = context.GetType().GetMethod(nameof(CreateVertexBuffer)).MakeGenericMethod(type).Invoke(context, new[] { t });
-							return typeof(ThreadedVertexBuffer<>).MakeGenericType(type).GetConstructors()[0].Invoke(new[] { this, vertexBuffer });
+							var vertexBuffer = context.GetType()
+								.GetMethod(nameof(CreateEmptyVertexBuffer))
```

Comment:
In general. All this introspection is slow. Similar code in TraitDictionary showed up as being slow. 


Isn't this too much abstraction for something that gets called often? Couldn't you hide it in some driver class that allocated a dedicated specific type of vertex buffer.

Reply:
This isn't called often. You'll create a vertex buffer and then reuse it.